lib/active_merchant/billing/gateways/pin.rb

```diff
@@ -149,23 +149,39 @@ def add_capture(post, options)
         post[:capture] = capture != false
       end
 
-      def add_creditcard(post, creditcard)
-        if creditcard.respond_to?(:number)
-          post[:card] ||= {}
+      def add_payment_method(post, payment_method)
+        return unless payment_method
 
+        case payment_method
+        when NetworkTokenizationCreditCard
+          post[:card] ||= {}
+          post[:card].merge!(
+            number: payment_method.number,
+            expiry_month: payment_method.month,
+            expiry_year: payment_method.year,
+            network_type: payment_method.source.to_s.gsub('_', ''),
+            cryptogram: payment_method.payment_cryptogram,
+            eci: payment_method.eci,
+            cvc: payment_method.verification_value,
+            name: payment_method.name
+          )
```

Comment:
@naashton this won't work, in order to make a charge with a network tokenized credit card it will need to be stored with the [payment sources API endpoint](https://pinpayments.com/developers/api-reference/payment-sources#type__network_token) before posting to the [charges API](https://pinpayments.com/developers/api-reference/charges#post-charges) with the resulting `payment_source_token`.

Reply:
@naashton the customers API doesn't currently support creating a customer with a payment source token, or network tokenized card, however, you can work around this with the following:

* Store a network tokenized card using the [payment sources API](https://pinpayments.com/developers/api-reference/payment-sources#type__network_token)
* Use the card token returned from the payment sources endpoint under `response.source.token` to create a customer
* You can then use the customer token to make a charge or activate a subscription

Reply:
Excellent, thanks for this clarification!